apps/SageAccountWeb/src/LoginPage.tsx

```diff
@@ -1,30 +1,64 @@
-import React from 'react'
 import { Box, Typography } from '@mui/material'
+import React from 'react'
+import { useHistory } from 'react-router-dom'
 import {
-  preparePostSSORedirect,
-  redirectAfterSSO,
+  LoginMethod,
   StandaloneLoginForm,
   SynapseConstants,
   SystemUseNotification,
+  preparePostSSORedirect,
+  redirectAfterSSO,
+  setLastLoginInfo,
   useApplicationSessionContext,
 } from 'synapse-react-client'
-import { SourceAppDescription, SourceAppLogo } from './components/SourceApp.js'
+import { backButtonSx } from './components/BackButton.js'
+import {
+  SourceAppDescription,
+  SourceAppLogo,
+  useSourceApp,
+} from './components/SourceApp.js'
 import {
   StyledInnerContainer,
   StyledOuterContainer,
 } from './components/StyledComponents.js'
-import { useHistory } from 'react-router-dom'
-import { backButtonSx } from './components/BackButton.js'
 
 export type LoginPageProps = {
   returnToUrl?: string
 }
 
+function formatDate(date: Date) {
+  const monthDay = date.toLocaleDateString('en-US', {
+    month: 'long',
+    day: 'numeric',
+  })
+  const year = date.toLocaleDateString('en-US', { year: 'numeric' })
+  return `${monthDay}, ${year}`
+}
+
+function getLoginMethodByProviderQueryParam(providerQueryParam: string | null) {
+  for (const [key, value] of Object.entries(
+    SynapseConstants.OAUTH2_PROVIDERS,
+  )) {
+    if (providerQueryParam === value) {
+      return key as LoginMethod
+    }
+  }
+  // if there isn't a provider query param, then assume logged in via email
+  return SynapseConstants.LOGIN_USERNAME_PASSWORD
+}
```

Comment:
Identify the login method using the provider query parameter. If there isn't a provider query parameter, assume that user logged in via email